napalm/base/test/helpers.py

```diff
@@ -4,23 +4,26 @@
 def test_model(model, data):
     """Return if the dictionary `data` complies with the `model`."""
     # Access the underlying schema for a TypedDict directly
-    model = model.__annotations__
-    same_keys = set(model.keys()) == set(data.keys())
+    annotations = model.__annotations__
+    if model.__total__:
+        same_keys = set(annotations.keys()) == set(data.keys())
+    else:
+        same_keys = set(data.keys()) <= set(annotations.keys())
 
     if not same_keys:
         print(
             "model_keys: {}\ndata_keys: {}".format(
-                sorted(model.keys()), sorted(data.keys())
+                sorted(annotations.keys()), sorted(data.keys())
             )
         )
 
     correct_class = True
-    for key, instance_class in model.items():
-        correct_class = isinstance(data[key], instance_class) and correct_class
+    for key in data.keys():
```

Comment:
Again we are switching over from the `annotations` to the `data` which is dangerous. Can we call out the `allow_subset=True` to specifically allow this in this case?